File: MHealthPlus/src/com/lucentinsight/mclinicplus/data/Clinic.java
package com.lucentinsight.mclinicplus.data;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

public class Clinic {

private String name;
private String address;
private double lat;
private double lang;
private String contactNo;
private Map<String, Doctor> doctors = new HashMap<String, Doctor>();
private Set<String> specializations = new HashSet<String>();

public void addSpecialization(String specialization){
	this.specializations.add(specialization);
}

public Set<String> getSpecialization() {
	return specializations;
}

public void setSpelications(Set<String> specialization) {
	this.specializations = specialization;
}

public Map<String, Doctor> getDoctors() {
	return doctors;
}

public void addDoctor(String specialization,Doctor doctor){
	doctors.put(specialization, doctor);
}
public void setDoctors(Map<String, Doctor> doctors) {
	this.doctors = doctors;
}
public String getName() {
	return name;
}
public void setName(String name) {
	this.name = name;
}
public String getAddress() {
	return address;
}
public void setAddress(String address) {
	this.address = address;
}
public double getLat() {
	return lat;
}
public void setLat(double lat) {
	this.lat = lat;
}
public double getLang() {
	return lang;
}
public void setLang(double lang) {
	this.lang = lang;
}
public String getContactNo() {
	return contactNo;
}
public void setContactNo(String contactNo) {
	this.contactNo = contactNo;
}


}

File: backend/mclincBackend/mclinicplus.com/php/entities/Clinic.php
<?php
/** @Entity
  * @Table(name="CP_CLINIC")
*/
class Clinic
{
/** @Id @Column(type="integer")
 * @GeneratedValue
 */
   private $id;

      public function getId(){
      return $this->id;
      }

	  public function setId($id){
	   $this->id=$id;
	   }
	   



   /** @Column(type="string") */
      private $name;

      public function getName(){
	        return $this->name;
	        }

	  	  public function setName($name){
	  	   $this->name=$name;
	   }
	   
	  
	   
	  /** @Column(type="string") */
      private $address;

      public function getAddress(){
	        return $this->address;
	  }

	  public function setAddress($address){
	  	   $this->address=$address;
	   } 
	   
  /** @Column(type="decimal") */
      private $lat;

      public function getLat(){
	        return $this->lat;
	  }

	  public function setLat($lat){
	  	   $this->lat=$lat;
	   } 
	   
	     /** @Column(type="decimal") */
	         private $lang;
	   
	         public function getLang(){
	   	        return $this->lang;
	   	  }
	   
	   	  public function setlang($lang){
	   	  	   $this->lang=$lang;
	   } 
	   
   /** @Column(type="string") */
      private $contactNo;

      public function getContactNo(){
	  return $this->contactNo;
	  }

	  public function setContactNo($contactNo){
	  $this->contactNo=$contactNo;
	   }

    private $doctors;
    
    public function addDoctor($doctor){
        $this->doctors[$doctor->getId()]=$doctor;
    }

public function __construct()
  {
    $this->$doctors=array(); 
  }
	   
 public function toJSON(){
	// return '{"name":"'.$this->name;
	 $ret ='{ "name":"'.$this->name.'","address":"'.$this->address.'","lat":"'.$this->lat.'","lang":"'.$this->lang.'","contactNo":"'.$this->contactNo.'","doctors":[{}';
 if(count($this->doctors)!=0){
 foreach($this->doctors as $doctor){
   $ret=$ret.','.$doctor->toJSON();
 }
 }
 $ret=$ret.']}';
     //'"}';
	 return $ret;
     }	      	   	     	      	     	     	       	    
}
?>